zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/znode/groupacl/TestFixupACL.java

```diff
@@ -29,59 +30,123 @@
     List<ACL> returnedList =
         PrepRequestProcessor.fixupACL(testPath, crossDomainComponentAuthInfo, aclList);
     Assert.assertEquals(1, returnedList.size());
-    Id returnedId = returnedList.get(0).getId();
-    Assert.assertEquals("x509", returnedId.getScheme());
-    Assert.assertEquals("crossDomain", returnedId.getId());
+    Assert.assertTrue(returnedList.contains(
+        new ACL(ZooDefs.Perms.ALL, new Id(X509AuthenticationUtil.X509_SCHEME, crossDomain))));
   }
 
   @Test
   public void testSuperUserId() throws KeeperException.InvalidACLException {
-    List<ACL> returnedList =
-        PrepRequestProcessor.fixupACL(testPath, superUserAuthInfo, aclList);
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, superUserAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
+    Assert.assertTrue(returnedList.containsAll(aclList));
+  }
+
+  @Test
+  public void testSingleDomainUser() throws KeeperException.InvalidACLException {
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, singleDomainAuthInfo, aclList);
     Assert.assertEquals(1, returnedList.size());
-    Id returnedId = returnedList.get(0).getId();
-    Assert.assertEquals("x509", returnedId.getScheme());
-    Assert.assertEquals("toBeAdded", returnedId.getId());
+    Assert
+        .assertTrue(returnedList.contains(new ACL(ZooDefs.Perms.ALL, singleDomainAuthInfo.get(0))));
+  }
+
+  @Test
+  public void testCnxnFiltering() throws KeeperException.InvalidACLException {
+    System.setProperty(X509AuthenticationConfig.DEDICATED_DOMAIN, dedicatedDomain);
+    List<ACL> returnedList =
+        PrepRequestProcessor.fixupACL(testPath, dedicatedDomainAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
+    Assert.assertTrue(returnedList.containsAll(aclList));
+    System.clearProperty(X509AuthenticationConfig.DEDICATED_DOMAIN);
+  }
+
+  @Test
+  public void testOpenReadPaths() throws KeeperException.InvalidACLException {
+    System.setProperty(X509AuthenticationConfig.OPEN_READ_ACCESS_PATH_PREFIX, testPath);
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, singleDomainAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
+    Assert.assertTrue(
+        returnedList.contains(new ACL(ZooDefs.Perms.READ, ZooDefs.Ids.ANYONE_ID_UNSAFE)));
+    System.clearProperty(X509AuthenticationConfig.OPEN_READ_ACCESS_PATH_PREFIX);
+  }
+
+  @Test
+  public void testMultiId() throws KeeperException.InvalidACLException {
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, multiIdAuthInfo, aclList);
+    Assert.assertEquals(2, returnedList.size());
+    multiIdAuthInfo
+        .forEach(id -> Assert.assertTrue(returnedList.contains(new ACL(ZooDefs.Perms.ALL, id))));
+  }
+
+  @Test
+  public void testNonX509ZnodeGroupAclUser() throws KeeperException.InvalidACLException {
+    List<ACL> returnedList = PrepRequestProcessor.fixupACL(testPath, nonX509AuthInfo, aclList);
```

Comment:
Nit: could we explain what we're doing and what the expected values should be in the `returnedList` for other developers who'll get to work on this in the future?